test/action_test.dart

```diff
@@ -35,30 +35,39 @@ void main() {
       expect(_action == _action2, isFalse);
     });
 
-    test('should support dispatch without a payload', () {
+    test('should support dispatch without a payload', () async {
+      Completer c = new Completer();
       Action _action = new Action();
 
-      _action.listen((payload) => expectAsync((payload) {
-            expect(payload, equals(null));
-          }));
+      _action.listen((String payload) {
+        expect(payload, equals(null));
+        c.complete();
+      });
 
       _action();
+      return c.future;
```

Comment:
does it make a difference as to whether you are returning the future rather than awaiting it?